src/charm.py

```diff
@@ -29,12 +30,34 @@
 KUBE_CONFIG_PATH = "/home/ubuntu/.kube/config"
 
 
-def determine_arch() -> str:
-    """Dpkg wrapper to surface the architecture we are tied to."""
-    cmd = ["dpkg", "--print-architecture"]
-    output = subprocess.check_output(cmd).decode("utf-8")
+class KubeConfigResourceManager:
+    """Manage the kubeconfig resource."""
 
-    return output.rstrip()
+    def __init__(self, model: ops.model.Model):
+        self.kube_config_path = Path(KUBE_CONFIG_PATH)
+        self.resource = model.resources.fetch("kubeconfig")
```

Comment:
check the ops docs... i think this CAN fail to fetch
https://ops.readthedocs.io/en/latest/#ops.Resources.fetch